### DISCUSSION TITLE
Vivado implementation fail & post-syn simulation fail

### DISCUSSION THREAD

**Discussion by junshiuanhsieh** [bol-edu/HLS-SOC-Discussions#99] [Q&A]
請問在 vivado synthesis完成後，需要跑implementation嗎，我跑implementation時出現以下的error，看起來是io的問題，請問這個該如何解決

![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/58873985/e3294fdf-0660-40b1-9ca8-faf0354c1496)

另外，請問跑完synthesis之後是否要跑post-synthesis simulation (有functional和timing)，我跑timing simulation時出現與functional不一樣的結果，想請問這是合成完有slack的意思嗎，還是有哪邊有可能有問題? 然後我也不太了解要如何在vivado裡面調整cycle time

**Comment by ZheChen-Bill** [comment]
從訊息上推斷問題出在板子的I/O port不夠所導致的問題，或許可以嘗試用不同的板子進行測試。

**Reply by ZheChen-Bill** [reply]
關於第二個問題，設定clock constraints後再進行synthesis，再選取report timing summary

**Reply by v00000v** [reply]
請問clock constraints要在哪裡設定呢?

**Comment by Lanxas1018** [comment]
同學好，這個Lab只需要做完synthesis就好，不需做implementation。
